### PR TITLE
Tableview crash bug

### PR DESCRIPTION
**Pull request by yihuiliao**
Might be related to https://github.com/adobe/react-spectrum/issues/5357

Some observations:
- Throttling the cpu seems to help prevent the crash
- Virtualizer is bouncing between rects
- Implementing a fix like the one here https://github.com/adobe/react-spectrum/pull/4981/files (with some minor changes) does help to solve the issue

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

1. Pull down branch
2. Run storybook locally
3. Go to TableView > Async Reload on Sort
4. Set window to 1082.73 width (or as close to it as possible) 90% zoom
5. Try sorting  the `Title` column by clicking on it
6. It should crash but you might need to click on `Title` a couple times to get it to work. See if you notice any difference with Strictmode on or off. 

Note: It is a bit finicky but both Rob and I have been able to reproduce it at this width and zoom pretty consistently.

### CODE REVIEW

**Comment by LFDanLu** [comment]
Note about the wrapping the following [updateSize](https://github.com/adobe/react-spectrum/blob/2bfb2a11f590e47ec9c158c8ef89c05c90fc8adc/packages/%40react-aria/virtualizer/src/ScrollView.tsx#L149-L151) in RAFs: it resolves the crashes in both TableView and CardView but infinite resize loops can still occur. They don't crash the components but they are visible as constant flickering at specific breakpoints unique to each setup

**Comment by snowystinger** [comment]
We've broken header in Tables, they no longer scroll with the body.
https://reactspectrum.blob.core.windows.net/reactspectrum/03c4a8b1b4e17c675c0c2320a3f3992115943e65/storybook/index.html?path=/story/tableview--empty-state-story&providerSwitcher-express=false

**Comment by devongovett** [comment]
I'd rather not add a new property to LayoutInfo. Instead, I think we can make the header have an inner div with the content size so it matches how ScrollView works.

File: packages/@react-spectrum/table/src/TableViewBase.tsx [hunk 1]
```diff
@@ -655,7 +656,14 @@ function TableVirtualizer(props) {
               transition: state.isAnimating ? `none ${state.virtualizer.transitionDuration}ms` : undefined
             }}
             ref={headerRef}>
-            {state.visibleViews[0]}
+            <div
```
Comment (snowystinger) [inline packages/@react-spectrum/table/src/TableViewBase.tsx:659]: clever 👍🏻

**Review by snowystinger — DISMISSED** [review]

**Review by LFDanLu — COMMENTED** [review]
The following stories now crash with the 185 error:
https://reactspectrum.blob.core.windows.net/reactspectrum/14a4cd3d812ae383add4677d7f50e1c15096664a/storybook/index.html?path=/story/tableview--typeahead-with-dialog&providerSwitcher-express=false
https://reactspectrum.blob.core.windows.net/reactspectrum/14a4cd3d812ae383add4677d7f50e1c15096664a/storybook/index.html?path=/story/tableview--links&providerSwitcher-express=false
https://reactspectrum.blob.core.windows.net/reactspectrum/14a4cd3d812ae383add4677d7f50e1c15096664a/storybook/index.html?path=/story/tableview--column-header-focus-ring-table&providerSwitcher-express=false

Only started happening once we changed to using the 100% width method, I don't really see any common pattern between those three at a glance, in fact the "column header focus after loading" seems pretty straightforward and has a set height...
Happy to approve if we still want to get a nightly for this

**Comment by snowystinger** [comment]
Oo, those all crash instantly, i don't think we want to introduce that

**Review by LFDanLu — APPROVED** [review]

**Review by dannify — APPROVED** [review]

**Pull request merged**